### DISCUSSION TITLE
Wrong assumption that args._[0] is a project directory or something like that

### DISCUSSION THREAD

**Discussion by 3axap4eHko** [vercel/next.js#36606] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #45-Ubuntu SMP Tue Mar 29 14:48:14 UTC 2022
    Binaries:
      Node: 16.15.0
      npm: 8.5.5
      Yarn: 1.22.15
      pnpm: 6.11.0
    Relevant packages:
      next: 12.1.5
      react: 18.1.0
      react-dom: 18.1.0

### What browser are you using? (if relevant)

N/A

### How are you deploying your application? (if relevant)

N/A

### Describe the Bug

I'm running the app under supervisor PM2 and once I start it, `next/bin/next` assumes that the first argument in `process.argv` is a command [see here](https://github.com/vercel/next.js/blob/canary/packages/next/bin/next.ts#L56). In the reality, the `argv` looks like this
```js
[
  '/path/to/bin/node',
  '/path/to/app/node_modules/pm2/lib/ProcessContainer.js',
  'ecosystem.config.js',
  'start'
]
```
So in this case, neither the command nor project directory is not what is expected to be, and nextjs throws an error that `/path/to/app/ecosystem.config.js/.env` is not found and running in dev mode

### Expected Behavior

expected to identify command and directory

### To Reproduce

Create a nextjs app and install pm2. Then create an `ecosystem.config.js` file
```js
module.exports = {
  apps: [
    {
      name: 'app',
      script: './node_modules/next/dist/bin/next',
      args: 'start'
    }
  ]
}
```
then start the app using `yarn pm2-runtime ecosystem.config.js`

**Comment by icyJoseph** [comment]
Ah, it took me a while to see what was different here. 

On my `package.json` I have:

```json
    "start": "next start",
```

And the `ecosystem.config.js`:

```js
module.exports = {
  apps: [
    {
      name: "next-site",
      cwd: " ./",
      script: "npm",
      args: "start",
    },
  ],
};
```

And then it all starts without issue. 

I'd say that your **_observation is still correct_** though, but it is kind of fragile to depend on the location of the `next` binary. Rather one should use the script commands.

**Reply by 3axap4eHko** [reply]
@icyJoseph  In this case, I should start npm to start pm2 to start npm to start next, is it the best solution we have now?

**Reply by 3axap4eHko** [reply]
BTW in this case PM2 controls not next app, but npm

**Reply by 3axap4eHko** [reply]
Also, if my observation is still correct, maybe it should be a bug?

**Reply by icyJoseph** [reply]
Yeah, you could alias `next start` to any name, "start" is just a convention.

I think this should also work with your setup though. Could you confirm?

```js
module.exports = {
  apps: [
    {
      name: "app",
      script: "./node_modules/.bin/next",
      args: "start",
    },
  ],
};
```

**Reply by 3axap4eHko** [reply]
@icyJoseph it does not work with yarn anyway as in my case

**Reply by 3axap4eHko** [reply]
it's weird, I just figured out that it does not work like this `NODE_ENV=production yarn start` maybe it's something with yarn even.

**Reply by 3axap4eHko** [reply]
`NODE_ENV=production npm run start` does not work too

**Reply by 3axap4eHko** [reply]
```bash
  errno: -20,
  syscall: 'stat',
  code: 'ENOTDIR',
  path: '/path/to/project/ecosystem.config.js/.env.development.local'
```

**Reply by 3axap4eHko** [reply]
but it works without leading environment variable

**Reply by icyJoseph** [reply]
Well this looks like the issue is something else? 

From those logs it seems like it's trying to load a file from a folder called `ecosystem.config.js`, and that's wrong right?

I am not sure what's `yarn start` or `npm run start` doing in your `package.json`, but this looks strange. When you do `next start`, NODE_ENV is set to production already.

**Reply by 3axap4eHko** [reply]
My start script is just `"start": "pm2-runtime ecosystem.config.js"`. I need NODE_ENV to be set `production` for `ecosystem.config.js` to understand how many clusters it needs to spawn and etc. Also, there is nothing wrong to set NODE_ENV before `next start`

**Reply by conanbatt** [reply]
I just got here with the same issue as the original author.

@icyJoseph the issue is that pm2-runtime makes the path change. They have a special comment on nuxtjs for it - https://pm2.keymetrics.io/docs/usage/docker-pm2-nodejs/

@3axap4eHko did you find a workaround for clustermode and pm2-runtime?
